src/aktualizr_info/aktualizr_info_test.cc

```diff
@@ -0,0 +1,478 @@
+  db_storage_->storeEcuRegistered();
+
+  Json::Value images_root_json;
+  images_root_json["key-001"] = "value-002";
+
+  std::string images_root = Utils::jsonToStr(images_root_json);
+  db_storage_->storeRoot(images_root, Uptane::RepositoryType::Image(), Uptane::Version(1));
+  db_storage_->storeRoot(images_root, Uptane::RepositoryType::Director(), Uptane::Version(1));
+
+  Json::Value image_targets_json;
+  image_targets_json["key-004"] = "value-005";
+  std::string image_targets_str = Utils::jsonToStr(image_targets_json);
+  db_storage_->storeNonRoot(image_targets_str, Uptane::RepositoryType::Image(), Uptane::Role::Targets());
+
+  SpawnProcess("--images-target");
+  ASSERT_FALSE(aktualizr_info_output.empty());
+
+  EXPECT_NE(aktualizr_info_output.find(device_id), std::string::npos);
+  EXPECT_NE(aktualizr_info_output.find(primary_ecu_serial), std::string::npos);
+  EXPECT_NE(aktualizr_info_output.find(primary_ecu_id), std::string::npos);
+
+  EXPECT_NE(aktualizr_info_output.find("image targets.json content:"), std::string::npos);
+  EXPECT_NE(aktualizr_info_output.find(image_targets_str), std::string::npos);
+}
+
+/**
+ * Verifies aktualizr-info output of a root metadata from the director repository
+ *
+ * Checks actions:
+ *
+ *  - [x] Print root metadata from director repository
+ */
+TEST_F(AktualizrInfoTest, PrintDirectorRootMetadata) {
+  db_storage_->storeEcuSerials({{Uptane::EcuSerial(primary_ecu_serial), Uptane::HardwareIdentifier(primary_ecu_id)}});
+  db_storage_->storeEcuRegistered();
+
+  Json::Value director_root_json;
+  director_root_json["key-002"] = "value-003";
+
+  std::string images_root = Utils::jsonToStr(director_root_json);
+  db_storage_->storeRoot(images_root, Uptane::RepositoryType::Director(), Uptane::Version(1));
+
+  SpawnProcess("--director-root");
+  ASSERT_FALSE(aktualizr_info_output.empty());
+
+  EXPECT_NE(aktualizr_info_output.find(device_id), std::string::npos);
+  EXPECT_NE(aktualizr_info_output.find(primary_ecu_serial), std::string::npos);
+  EXPECT_NE(aktualizr_info_output.find(primary_ecu_id), std::string::npos);
+
+  EXPECT_NE(aktualizr_info_output.find("director root.json content:"), std::string::npos);
+  EXPECT_NE(aktualizr_info_output.find(images_root), std::string::npos);
+}
+
+/**
+ * Verifies aktualizr-info output of targets' metadata from the director repository
+ *
+ * Checks actions:
+ *
+ *  - [x] Print targets metadata from director repository
+ */
+TEST_F(AktualizrInfoTest, PrintDirectorTargetsMetadata) {
+  db_storage_->storeEcuSerials({{Uptane::EcuSerial(primary_ecu_serial), Uptane::HardwareIdentifier(primary_ecu_id)}});
+  db_storage_->storeEcuRegistered();
+
+  Json::Value director_root_json;
+  director_root_json["key-002"] = "value-003";
+
+  std::string images_root = Utils::jsonToStr(director_root_json);
+  db_storage_->storeRoot(images_root, Uptane::RepositoryType::Director(), Uptane::Version(1));
```

Comment:
Why store the `images_root` here?

Reply:
Because, the aktualizr-info won't output target metadata from a director repository if its root metadata are not stored in DB. Although, the variable name 'images_root' is incorrect so I am renaming it into 'director_root'.

Reply:
Great. now I understand. The name threw me off, so thanks for fixing it.